frontend/apps/filemanager/filemanager.lua

```diff
@@ -539,6 +539,15 @@ function FileManager:init()
 
     self:initGesListener()
     self:handleEvent(Event:new("SetDimensions", self.dimen))
+
+    -- NOTE: ReaderUI has a _getRunningInstance method for this, because it used to store the instance reference in a private module variable.
+    if FileManager.instance == nil then
+        logger.dbg("Spinning up new FileManager instance", tostring(self))
+    else
+        -- Should never happen, given what we did above...
+        logger.warn("FileManager instance mismatch! Opened", tostring(self), "while we still have an existing instance:", tostring(FileManager.instance))
```

Comment:
You could use a dbg.err or something to make sure we actually notice it.

Reply:
@NiLuJe I actually meant the straight up crash dbg.dassert variety like:

https://github.com/koreader/koreader/blob/ea3fa5c2c7ed7330e8e1e6ef838b780e7998ea51/plugins/batterystat.koplugin/main.lua#L244

Reply:
Huh. Missed that one when going through Dbg (which, err, could use some documentation ^^).

Reply:
I was being rather vague I suppose. I mainly meant anything along the lines of dbg.guard/dbg.dassert/dbg.whatever that you feel is appropriate but hard crashes the program so I would actually notice if or when it happens. ;-)

At the same time this isn't a very important issue, hopefully.

Reply:
For future reference: https://github.com/koreader/koreader/pull/7677

Reply:
Yeah, that one branch should *really* be dead code ;).